tools/clang/include/clang/Basic/DiagnosticIDs.h

```diff
@@ -36,7 +36,8 @@ namespace clang {
       DIAG_START_AST           = DIAG_START_PARSE           +  500,
       DIAG_START_COMMENT       = DIAG_START_AST             +  110,
       DIAG_START_SEMA          = DIAG_START_COMMENT         +  100,
-      DIAG_START_ANALYSIS      = DIAG_START_SEMA            + 3000,
+      // HLSL Change: SEMA group length increased from 3000.
+      DIAG_START_ANALYSIS      = DIAG_START_SEMA            + 3100,
```

Comment:
Oh great, I touched a file (not ours) that doesn't currently pass clang-format check :(

What should I do?  clang-format the file?  I don't like that option.

Here's an example where the clang-formatted code is harder to read, too.
Is this really better?
```cpp
  enum {
    DIAG_START_COMMON = 0,
    DIAG_START_DRIVER = DIAG_START_COMMON + 300,
    DIAG_START_FRONTEND = DIAG_START_DRIVER + 100,
    DIAG_START_SERIALIZATION = DIAG_START_FRONTEND + 100,
    DIAG_START_LEX = DIAG_START_SERIALIZATION + 120,
    DIAG_START_PARSE = DIAG_START_LEX + 300,
    DIAG_START_AST = DIAG_START_PARSE + 500,
    DIAG_START_COMMENT = DIAG_START_AST + 110,
    DIAG_START_SEMA = DIAG_START_COMMENT + 100,
    // HLSL Change: SEMA group length increased from 3000.
    DIAG_START_ANALYSIS = DIAG_START_SEMA + 3100,
    DIAG_UPPER_LIMIT = DIAG_START_ANALYSIS + 100
  };
```

Reply:
If I start reformatting from here, the whole file will need reformatting, since it doesn't like the tabbed-in namespaces.  I really don't want to reformat the file as part of this change.  It really doesn't make sense.  I don't think we should reformat the whole file at all.

Reply:
In this case I think it would be best to override the PR check for clang format if this is the only issue.

Reply:
I concur

Reply:
Considering the rate at which we've been adding Sema Diagnostics, 100 more seems kind of low. Is there a reason not to add 1000?